fdbserver/VersionedBTree.actor.cpp

```diff
@@ -5717,27 +5878,42 @@ class VersionedBTree {
 		}
 
 		PageToBuild p(0, m_blockSize, m_encodingType);
+		bool beforeAddNewRecord = true;
```

Comment:
With debug printf disabled, does this give an unused variable warning?  It is set, but the only read I see is the check before a `debug_printf` which would would preprocess to nothing so the conditional would be removed.

Reply:
Also I think there may be a way to detect what you are looking for here from `p` itself.